README.md

```diff
@@ -230,6 +230,34 @@ the `@optional` does not apply, and that value is then checked against the filte
 coercion. These subsequent operations may then cause the result set to be discarded if it does
 not match.
 
+For example, suppose we have two Person nodes with names `Albert` and `Betty` such that there's an edge 
+`Albert-knows->Betty`.
+
+Then for the following query:
+```graphql
+{
+  Person {
+    out_Person_Knows @optional {
+      name @filter(op_name: "=", value: "Charles")
+    }
+    name @output(out_name: "person_name")
+  }
+}
+```
+then the output would be 
+```graphql
+{}
+```
+because the `Albert-knows->Betty` edge satisfies the `@optional` directive, but this edge is then discarded because it doesn't match the filter checking if Betty's name is Charles. 
```

Comment:
This part of the sentence is not technically correct. For instance, the result set is discarded not the edge. I am having trouble figuring out how to rephrase this. @obi1kenobi what do you think?

Reply:
@obi1kenobi ?

Reply:
How's this: 

because the `Person_Knows` edge from `Albert` to `Betty` satisfies the `@optional` directive, but `Betty` doesn't match the filter checking for a node with name `Charles`.

Reply:
Ok sounds good.